datafusion/src/physical_plan/expressions/min_max.rs

```diff
@@ -129,11 +131,49 @@ macro_rules! typed_min_max_batch {
     }};
 }
 
+// TODO implement this in arrow-rs with simd
+// https://github.com/apache/arrow-rs/issues/1010
+// Statically-typed version of min/max(array) -> ScalarValue for decimal types.
+macro_rules! typed_min_max_batch_decimal128 {
+    ($VALUES:expr, $PRECISION:ident, $SCALE:ident, $OP:ident) => {{
+        let null_count = $VALUES.null_count();
+        if null_count == $VALUES.len() {
+            ScalarValue::Decimal128(None, *$PRECISION, *$SCALE)
+        } else {
+            let array = $VALUES.as_any().downcast_ref::<DecimalArray>().unwrap();
+            if null_count == 0 {
+                // there is no null value
+                let mut result = array.value(0);
+                for i in 1..array.len() {
+                    result = result.$OP(array.value(i));
+                }
+                ScalarValue::Decimal128(Some(result), *$PRECISION, *$SCALE)
+            } else {
+                let mut result = 0_i128;
```

Comment:
It might be more idomatic to use `let mut rust: Option<i128> = 0;` (aka use an `Option` rather than explicit flag)

Then instead of code like

```rust
                    if !has_value && array.is_valid(i) {
                        has_value = true;
                        result = array.value(i);
                    }
                    if array.is_valid(i) {
                        result = result.$OP(array.value(i));
                    }
```

You could write code like the following (which I think saves at least one check of `is_valid()`):

```rust
                    if array.is_valid(i) {
                        let value = array.value(i);
                        result = result.$OP(result.unwrap_or(value)
                    }
```

This is just a style suggestion, it is not needed I don't think

Reply:
I plan to resolve it in the follow-up pull request. 
You can merge it if it looks good to you.

Reply:
I recheck your suggestion and find some bugs.
For example, if all value in the array is less than zero, it may be [-1,-3,-3] and we want to get the max value of them.
initially, we set the result as `Some(0)`  and follow your suggestion code, we will get the `0` as the max value for the result.

I think we should can just change to that
```
let mut result = 0_i128; ->>> let mut result : i128 = 0;
```
In addition the `unwrap_or` is 
```
    pub fn unwrap_or(self, default: T) -> T {
        match self {
            Some(x) => x,
            None => default,
        }
    }
```